modules/system/assets/js/framework.js

```diff
@@ -91,6 +91,27 @@ if (window.jQuery.request !== undefined) {
             inputName,
             data = {}
 
+        // options.data contains $el's data-request-data
+
+        // Data empty
+
+        // Data contains data-request-data data included by any element that is a parent of $el (all the way up to
+        // the document root) in closest-takes-priority scheme
+
+        // If $el is a form input AND is not present in a $form container then options.data.{$el.name}
+        // is overwritten with $el.value
+
+        // If options.data is a valid object then merge data & options.data prioritizing options.data
+
+        // If the containing form has a request-parent element specified, then we move to that element
+        // and process it's data as if it was the element creating the request, including checking for a
+        // parent element to that request. Once we've gone all the way up until there are no more parent
+        // elements detected then the parent data is merged such that the closest data in the chain to the
+        // original element that triggered the request has the highest priority.
+
+
+        // Loop through every parent element with the data-request-data attribute and attach the data
+        // to the request data object with closer element data overwriting further element data
```

Comment:
@LukeTowers is this comment needed when we've got an explanatory comment further down, in the `getParentFormData` method?

Reply:
Probably not, it was more for my own reference; I'll go through and clean it up tomorrow